sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Combine.java

```diff
@@ -1890,17 +1887,21 @@ public void populateDisplayData(DisplayData.Builder builder) {
     private final DisplayData.ItemSpec<? extends Class<?>> fnDisplayData;
     private final SerializableFunction<? super K, Integer> hotKeyFanout;
 
-    private PerKeyWithHotKeyFanout(String name,
+    private PerKeyWithHotKeyFanout(
         PerKeyCombineFn<? super K, ? super InputT, ?, OutputT> fn,
         DisplayData.ItemSpec<? extends Class<?>> fnDisplayData,
         SerializableFunction<? super K, Integer> hotKeyFanout) {
-      super(name);
       this.fn = fn;
       this.fnDisplayData = fnDisplayData;
       this.hotKeyFanout = hotKeyFanout;
     }
 
     @Override
+    protected String getKindString() {
+      return String.format("Combine.perKeyWithFanout(%s)", NameUtils.approximateSimpleName(fn));
```

Comment:
There is some amount cross-language consistency to kind strings being of the grammar:

_kindString_ ::= _expression_
_expression_ ::= _symbol args?_
_symbol_ ::= _capitalizedAtom (_`.` _symbol)*_
_args_ ::= `(` _expression (_ `,` _expression)*_ `)`

This works in a lot of languages and is just sort of a basically mathematical notation with the addition of dots. I get the impression this is how kind strings are laid out. But lets merge your PR as-is and just move forwards since they'll be obsoleted either way.